### PR TITLE
New Privacy Policy

### PR DESCRIPTION
**Pull request by oskarabc**
(test)

Fixes CSB-41

### CODE REVIEW

**Comment by lbogdan** [comment]
Build for latest commit https://github.com/codesandbox/codesandbox-client/pull/3815/commits/a05065ae56a7e9ba82dcf9176470cce2aede6885 is at https://pr3815.build.csb.dev/s/new.

File: packages/homepage/content/legal/privacy.md [hunk 1]
```diff
@@ -0,0 +1,322 @@
...
+
+This policy is further intended to apply to anyone exposed to any CodeSandbox
+service. To further clarify the terms we are using and how we are thinking,
+please refer to the glossary at the end of this document.
+
+Please reach out to us via our contact details below as soon as possible if you
+or anyone in your company using CodeSandbox feels uninformed, finds anything on
+our websites to be unfair, incompletely explained or surprising.
+
+## **In short**
```
Comment (MichaelDeBoey) [inline packages/homepage/content/legal/privacy.md:30]:
```suggestion
## In short
```

File: packages/homepage/content/legal/privacy.md [hunk 2]
```diff
@@ -0,0 +1,322 @@
...
+  Netlify deploy option and to improve what we are building by way of placing
+  tracking cookies;
+- We may promote our services to inform you via email about new releases,
+  community activities or for events that may ultimately affect your day-to-day
+  use of our services;
+- We may enrich your (personal) data with publicly accessible data;
+- We are operationally based in The Netherlands and therefore you are dealing
+  with applicable Dutch and/or EU law.
+
+## **What does CodeSandbox collect and why?**
```
Comment (MichaelDeBoey) [inline packages/homepage/content/legal/privacy.md:54]:
```suggestion
## What does CodeSandbox collect and why?
```

File: packages/homepage/content/legal/privacy.md [hunk 3]
```diff
@@ -0,0 +1,322 @@
...
+
+Personal data is collected via various sources on anyone exposed to any
+CodeSandbox service. Typically, users who create an account (as defined below)
+could provide more data. For examples, see the data types section below. You can
+limit or change the amount of personal data we process and store by limiting the
+data you provide to us in your account settings or by reaching out to us.
+
+### **Why we are collecting data**
```
Comment (MichaelDeBoey) [inline packages/homepage/content/legal/privacy.md:62]:
```suggestion
### Why we are collecting data
```

File: packages/homepage/content/legal/privacy.md [hunk 4]
```diff
@@ -0,0 +1,322 @@
...
+- Contact you with requested support;
+- Send you documentation, educational material related to our services;
+- Notify you of major upcoming product or legal policy changes;
+- Improve, research and develop new features as part of the services you may be
+  using;
+- Invite you to participate in surveys, where necessary subject to your consent;
+- Prevent misuse of or stop other illegal activities from being conducted
+  through our services.
+
+### **Data types we are collecting and using**
```
Comment (MichaelDeBoey) [inline packages/homepage/content/legal/privacy.md:90]:
```suggestion
### Data types we are collecting and using
```

File: packages/homepage/content/legal/privacy.md [hunk 5]
```diff
@@ -0,0 +1,322 @@
...
+providers with related tips.
+
+Financial. We use Stripe as a payment service provider and do not collect nor
+store credit card or other financial data in our databases. We do have access to
+Stripe's database for the purpose of complying with support requests and
+administration requirements, but no level of access exposes full credit card
+details. Only the last four digits can be read and are typically used for
+verification purposes during support interactions.
+
+### **Use of third parties, sharing data and sub-processors**
```
Comment (MichaelDeBoey) [inline packages/homepage/content/legal/privacy.md:141]:
```suggestion
### Use of third parties, sharing data and sub-processors
```

File: packages/homepage/content/legal/privacy.md [hunk 6]
```diff
@@ -0,0 +1,322 @@
...
+example of how hard it can be to discern who is delivering what service to you.
+Intuitive design has to be obfuscating in this regard and can ignite distrust in
+some, pleasant surprise in others.
+
+We have certain practices in place to ensure our partners adhere to legal
+standards and we typically interview new partners to get a sense of how they are
+treating user data. We have Data Processing Agreements in place with our
+partners and our sub-processors.
+
+### **Sub-processors & Data locations**
```
Comment (MichaelDeBoey) [inline packages/homepage/content/legal/privacy.md:175]:
```suggestion
### Sub-processors & Data locations
```

File: packages/homepage/content/legal/privacy.md [hunk 7]
```diff
@@ -0,0 +1,322 @@
...
+  Provider. [https://mailchimp.com/gdpr](https://mailchimp.com/gdpr).
+
+Your personal data may be transferred, stored and processed in the European
+Economic Area (“EEA”), United States (“US”) or any other country in which our
+service providers maintain facilities. By using our domains, you consent to any
+transfer, storing or processing of personal data outside of your country of
+residence and outside the EEA. We will take all steps reasonably necessary to
+ensure that your data is treated securely and in accordance with this policy.
+
+### **Retaining data**
```
Comment (MichaelDeBoey) [inline packages/homepage/content/legal/privacy.md:210]:
```suggestion
### Retaining data
```

File: packages/homepage/content/legal/privacy.md [hunk 8]
```diff
@@ -0,0 +1,322 @@
...
+
+We will not retain your personal data for a period longer than necessary to
+fulfill the purposes described in this policy, unless we have to keep it for
+legitimate tax, business or legal purposes.
+
+### **Securing data**
```
Comment (MichaelDeBoey) [inline packages/homepage/content/legal/privacy.md:216]:
```suggestion
### Securing data
```

File: packages/homepage/content/legal/privacy.md [hunk 9]
```diff
@@ -0,0 +1,322 @@
...
+  where it is appropriate to their tasks and/or roles and on a need-to-know
+  basis;
+- We obtain certifications to test our practices against public or industry
+  standards;
+- When we receive reports of abuse, data breaches pertaining to the integrity of
+  our users or our own, investigation immediately follows upon learning about it
+  and reasonable action is taken as swiftly as possible in accordance with
+  applicable privacy laws.
+
+## **Your rights and how to contact us**
```
Comment (MichaelDeBoey) [inline packages/homepage/content/legal/privacy.md:234]:
```suggestion
## Your rights and how to contact us
```

File: packages/homepage/content/legal/privacy.md [hunk 10]
```diff
@@ -0,0 +1,322 @@
...
+  to:
+  [https://edpb.europa.eu/about-edpb/board/members_en](https://edpb.europa.eu/about-edpb/board/members_en).
+- You have the right not to be subject to a decision based solely on automated
+  processing, including profiling, which adversely affects your legal rights or
+  substantially impairs you in a similar manner.
+
+Some of these rights may be executed with a click of a button from our website,
+insofar as they are not please reach out to us by the following means.
+
+### **Contact us**
```
Comment (MichaelDeBoey) [inline packages/homepage/content/legal/privacy.md:268]:
```suggestion
### Contact us
```

File: packages/homepage/content/legal/privacy.md [hunk 11]
```diff
@@ -0,0 +1,322 @@
...
+operational headquarters at Singel 542, 1017 AZ, Amsterdam, The Netherlands,
+available at +1 (650) 731-3185‬. To exercise any of the rights described above
+or in case of concerns, questions or inquiries, please email us at
+hello@codesandbox.io.
+
+We promise to respond as soon as possible and in any case within fourteen (14)
+business days of your request. For account deletion or data dumps, please
+mention "privacy" anywhere in the subject header.
+
+### **Age**
```
Comment (MichaelDeBoey) [inline packages/homepage/content/legal/privacy.md:280]:
```suggestion
### Age
```

File: packages/homepage/content/legal/privacy.md [hunk 12]
```diff
@@ -0,0 +1,322 @@
...
+
+Our services are intended for anyone at any age who would like to experiment
+with and learn about design and code. However, if you are under the age of 16,
+please do not provide us with any personal information, have your parents or
+legal guardian reach out to us immediately with permission or questions. Please
+also be aware we have no ready access or instant knowledge of anyone’s age.
+
+### **Applicable law**
```
Comment (MichaelDeBoey) [inline packages/homepage/content/legal/privacy.md:288]:
```suggestion
### Applicable law
```

File: packages/homepage/content/legal/privacy.md [hunk 13]
```diff
@@ -0,0 +1,322 @@
...
+
+CodeSandbox operates globally and you may have different rights under your local
+laws. We shall strive toward complying with laws beyond where we are based. This
+policy is construed under Dutch law. All privacy related disputes shall be
+exclusively submitted to a competent court in The Netherlands.
+
+## **Glossary**
```
Comment (MichaelDeBoey) [inline packages/homepage/content/legal/privacy.md:295]:
```suggestion
## Glossary
```

**Comment by DannyRuchtie** [comment]
-[] Add in app notification
-[] Close and never show again logic (@Sara ?)

File: packages/homepage/src/components/Toast/index.js [hunk 14]
```diff
@@ -0,0 +1,14 @@
+import React from 'react';
+import { Link } from 'gatsby';
+
+import { Toastcontainer } from './elements';
+
+const Privacy = () => (
```
Comment (MichaelDeBoey) [inline packages/homepage/src/components/Toast/index.js:6]: Make the component a `FunctionComponent` & export it as a named export in order to be the same as all our new components.

```suggestion
export const Privacy: FunctionComponent = () => (
```

File: packages/homepage/src/components/Toast/index.js [hunk 15]
```diff
@@ -0,0 +1,14 @@
+import React from 'react';
```
Comment (MichaelDeBoey) [inline packages/homepage/src/components/Toast/index.js:1]:
```suggestion
import React, { FunctionComponent } from 'react';
```

File: packages/homepage/src/components/Toast/index.js [hunk 16]
```diff
@@ -0,0 +1,14 @@
...
+
+const Privacy = () => (
+  <Toastcontainer>
+    By using our site you agree with our{' '}
+    <Link to="/legal/privacy">Privacy Policy</Link>
+    <span>close</span>
+  </Toastcontainer>
+);
+
+export default Privacy;
```
Comment (MichaelDeBoey) [inline packages/homepage/src/components/Toast/index.js:14]:
```suggestion
```

File: packages/homepage/src/pages/index.js [hunk 17]
```diff
@@ -3,7 +3,13 @@ import { identify } from '@codesandbox/common/lib/utils/analytics';
 
 import TitleAndMetaTags from '../components/TitleAndMetaTags';
 import Layout, { WRAPPER_STYLING } from '../components/layout';
+
+
+import Privacy from '../components/Toast';
+
```
Comment (MichaelDeBoey) [inline packages/homepage/src/pages/index.js:9]:
```suggestion
import { Privacy } from '../components/Toast';
```

**Comment by DannyRuchtie** [comment]
Thx @MichaelDeBoey for the help

File: packages/homepage/src/components/Toast/index.js [hunk 18]
```diff
@@ -0,0 +1,39 @@
+/* eslint-disable jsx-a11y/control-has-associated-label */
+import React, { useState, useEffect } from 'react';
+import { Link } from 'gatsby';
+import { AnimatePresence } from 'framer-motion';
+
+import { ToastContainer } from './elements';
+
+const Privacy = () => {
```
Comment (MichaelDeBoey) [inline packages/homepage/src/components/Toast/index.js:8]:
```suggestion
export const Privacy: FunctionComponent = () => {
```

File: packages/homepage/src/components/Toast/index.js [hunk 19]
```diff
@@ -0,0 +1,39 @@
+/* eslint-disable jsx-a11y/control-has-associated-label */
+import React, { useState, useEffect } from 'react';
```
Comment (MichaelDeBoey) [inline packages/homepage/src/components/Toast/index.js:2]:
```suggestion
import React, { FunctionComponent, useEffect, useState } from 'react';
```

File: packages/homepage/src/components/Toast/index.js [hunk 20]
```diff
@@ -0,0 +1,39 @@
...
+            ariaLabel="Close"
+            onClick={() => setShow(false)}
+          />
+        </ToastContainer>
+      ) : null}
+    </AnimatePresence>
+  );
+};
+
+export default Privacy;
```
Comment (MichaelDeBoey) [inline packages/homepage/src/components/Toast/index.js:39]:
```suggestion
```

File: packages/homepage/src/components/layout.js [hunk 21]
```diff
@@ -3,6 +3,7 @@ import { ThemeProvider } from 'styled-components';
 
 import theme from '@codesandbox/common/lib/theme';
 import { VisuallyHidden } from './style';
+import Privacy from './Toast';
```
Comment (MichaelDeBoey) [inline packages/homepage/src/components/layout.js:6]:
```suggestion
import { Privacy } from './Toast';
```

File: packages/homepage/src/components/Toast/index.js (hunk 18 above)
Comment (SaraVieira) [reply packages/homepage/src/components/Toast/index.js:8]: We are not using named exports or Typpescript in the homepage

File: packages/app/src/app/overmind/onInitialize.ts [hunk 22]
```diff
@@ -104,4 +106,26 @@ export const onInitialize: OnInitialize = async (
...
+            label: 'Open Privacy Policy',
+            run: () => {
+              window.open('https://codesandbox.io/legal/privacy', '_blank');
+            },
+          },
+        ],
+      },
+    });
+    effects.browser.storage.set(seenTermsKey, true);
+  }
```
Comment (MichaelDeBoey) [inline packages/app/src/app/overmind/onInitialize.ts:130]: Use early return instead

```suggestion
  if (effects.browser.storage.get(seenTermsKey)) {
    return;
  }
  
  effects.notificationToast.add({
    message:
      'Hello, our privacy policy has been updated recently. What’s new? CodeSandbox emails. Please read and reach out.',
    title: 'Updated Privacy',
    status: convertTypeToStatus('notice'),
    sticky: true,
    actions: {
      primary: [
        {
          label: 'Open Privacy Policy',
          run: () => window.open('https://codesandbox.io/legal/privacy', '_blank'),
        },
      ],
    },
  });
  effects.browser.storage.set(seenTermsKey, true);
```

File: packages/homepage/src/components/Toast/index.js (hunk 18 above)
Comment (MichaelDeBoey) [reply packages/homepage/src/components/Toast/index.js:8]: Doh my bad! 😓

File: packages/homepage/src/components/Toast/index.js [hunk 23]
```diff
@@ -0,0 +1,39 @@
+/* eslint-disable jsx-a11y/control-has-associated-label */
+import React, { useState, useEffect } from 'react';
+import { Link } from 'gatsby';
+import { AnimatePresence } from 'framer-motion';
+
+import { ToastContainer } from './elements';
+
+const Privacy = () => {
+  const key = 'ACCEPTED_TERMS_CODESANDBOX_HOME';
+  const [show, setShow] = useState(false);
```
Comment (MichaelDeBoey) [inline packages/homepage/src/components/Toast/index.js:10]: You can use a callback function to set initial value.

```suggestion
  const [show, setShow] = useState(() => !window.localStorage.getItem(key));
```

File: packages/homepage/src/components/Toast/index.js [hunk 24]
```diff
@@ -0,0 +1,39 @@
...
+
+  useEffect(() => {
+    setShow(!window.localStorage.getItem(key));
```
Comment (MichaelDeBoey) [inline packages/homepage/src/components/Toast/index.js:13]:
```suggestion
```

**Comment by oskarabc** [comment]
Major changes to our currently outdated privacy policy containing inaccurate statements. Main reason for the rewrites are:
* Add European or EEA focus and compliance with laws and regulations;
* Support introduction of upcoming email campaigns;
* Tell you who we partner with;
* Provide more accurate information on how we treat personal data.

This PR also introduces a homepage notification for visitors and a regular editor notification announcing this update.

Privacy is an ongoing concern for CodeSandbox and our next update is likely to contain more information on international relations, such as there is the Privacy Shield Framework between US-EU. Please do reach out to us with questions.

**Pull request merged**